tsfresh/feature_extraction/extraction.py

```diff
@@ -193,56 +200,57 @@ def _do_extraction(df, column_id, column_value, column_kind,
     :param column_value: The name for the column keeping the value itself.
     :type column_value: str
 
-    :param chunksize: The size of one chunk for the parallelization
-    :type chunksize: None or int
+    :param chunk_size: The size of one chunk for the parallelization
+    :type chunk_size: None or int
 
     :param n_jobs: The number of processes to use for parallelization. If zero, no parallelization is used.
     :type n_jobs: int
 
     :param disable_progressbar: Do not show a progressbar while doing the calculation.
     :type disable_progressbar: bool
 
+    :param distributor: Advanced parameter: set this to an instance name that you want to use as a
+             distributor. See the utilities/distribution.py for more information. Leave to None, if you want
+             TSFresh to choose the best distributor.
+    :type distributor: Distributor
+
     :return: the extracted features
     :rtype: pd.DataFrame
     """
     data_in_chunks = [x + (y,) for x, y in df.groupby([column_id, column_kind])[column_value]]
 
-    total_number_of_expected_results = len(data_in_chunks)
+    if distributor is None:
 
-    if n_jobs == 0:
-        map_function = map
-    else:
-        pool = Pool(n_jobs)
+        if n_jobs == 0:
+            distributor_class = MapDistributor
+        else:
+            distributor_class = MultiprocessingDistributor
 
-        if not chunksize:
-            chunksize = _calculate_best_chunksize(data_in_chunks, n_jobs)
+        distributor = distributor_class(n_workers=n_jobs, disable_progressbar=disable_progressbar,
+                                        progressbar_title="Feature Extraction")
 
-        map_function = partial(pool.imap_unordered, chunksize=chunksize)
+    elif isinstance(distributor, six.string_types):
 
-    extraction_function = partial(_do_extraction_on_chunk,
-                                  default_fc_parameters=default_fc_parameters,
-                                  kind_to_fc_parameters=kind_to_fc_parameters)
+        if is_valid_ip_and_port(distributor):
+            distributor = ClusterDaskDistributor(n_workers=n_jobs,  disable_progressbar=disable_progressbar,
+                                                 progressbar_title="Feature Extraction", address=distributor)
+        else:
+            raise ValueError(distributor + " is not a valid ip address")
 
-    # Map over all those chunks and extract the features on them
-    result = tqdm(map_function(extraction_function, data_in_chunks),
-                  total=total_number_of_expected_results,
-                  desc="Feature Extraction", disable=disable_progressbar)
+    if not isinstance(distributor, Distributor):
+        raise ValueError("the passed distributor is neither None nor an IP address or a Distributor object")
```

Comment:
Actually, I do not like mixing distributor instances and strings as an argument.
I would prefer that if users want a local dask cluster (which is somehow advance anyway), they just pass their own already created instance...

Reply:
It makes sense from the user perspective. I am still working on this branch. Give me a few days ;)

Reply:
Sure, just wanted to give my opinion :-)